Pinta.Core/Classes/PaintBrushOption/IntegerOption.cs

```diff
@@ -0,0 +1,58 @@
+//
+// IntegerOption.cs
+//
+// Author:
+//       Paul Korecky <https://github.com/spaghetti22>
+//
+// Copyright (c) 2025 Paul Korecky
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+using Gtk;
+
+namespace Pinta.Core;
+
+/// <summary>
+/// An integer-valued option that creates a spin button with a set minimum and maximum.
+/// </summary>
+public class IntegerOption : PaintBrushOption
```

Comment:
These aren't necessarily specific to brushes - maybe these could be useful to generalize more for any tool to use too?

From a maintenance perspective it would be great to reduce / eliminate tools having GTK dependencies for their toolbar interface, since I've had to upgrade a lot of code from GTK 3 -> 4 and eventually 5 in the future :)

We also did something along these lines already for effects, but it took a different approach of generating the widgets automatically by reflection on an EffectData struct (see the `SimpleEffectDialog` and any of the effects for an example)